src/worker/main.ts

```diff
@@ -1,24 +1,65 @@
 import { readExcel } from "./read-wordlist";
-import { SaveFileEventPayload } from "@common/events";
+import {
+  EventPayload,
+  SaveFileRequestPayload,
+  CompileModelRequestPayload,
+  CompileModelResponsePayload,
+  EventType,
+} from "@common/events";
+import {
+  compileModelFromLexicalModelSource,
+  WordListFromArray,
+} from "@predictive-text-studio/lexical-model-compiler";
 import Storage from "./storage";
 
 const storage = new Storage();
 
-// Print the function's name (even though that's self-evident) just to get the
-// linter and Rollup to acknowledge that readExcelSync() is used.
-self.postMessage(`hello, from worker! I can use ${readExcel.name}`);
-
-const handleSaveFileEvent = async (event: MessageEvent) => {
-  const payload = event.data as SaveFileEventPayload;
+const handleEvent = async (event: MessageEvent) => {
+  const payload = event.data as EventPayload;
+  switch (payload.type) {
+    case EventType.SaveFileRequest: {
+      await handleSaveFileEvent(payload as SaveFileRequestPayload);
+    }
+    case EventType.CompileModelRequest: {
+      await handleCompileModel(payload as CompileModelRequestPayload);
```

Comment:
A few things!

Every `switch/case` must have a `default`. Also, I don't assume you intended a fallthrough here, so I added break.
Why did you use `await` here? What purpose does it serve if this is a void function, and nothing `await`s `handleEvent()`'s return?

```suggestion
      handleSaveFileEvent(payload as SaveFileRequestPayload);
      break;
    }
    case EventType.CompileModelRequest: {
      handleCompileModel(payload as CompileModelRequestPayload);
      break;
    }
    default: {
       // TODO?????
```

Reply:
> Thank you for taking time to do this! However, I'm honestly not sure about this :/
> 
> It kind of touches upon #74.
> 
> So we could accept the changes here, and then focus _exclusively_ on refactoring and tests for the few PRs.
> 
> I would also like to see tests for this. Does this actually... you know... work? 
> 
> So how would you like to continue @jienius? Would you like to get this merged and then focus our attention next to testing, documenting, and refactoring? Or would you rather get that over with right now?
> 
> I feel like we're dragging on our "proof-of-concept" period, and I'd like to see more "production-ready" code. This means tests, documentation, and better factoring!

Cool

> Thank you for taking time to do this! However, I'm honestly not sure about this :/
> 
> It kind of touches upon #74.
> 
> So we could accept the changes here, and then focus _exclusively_ on refactoring and tests for the few PRs.
> 
> I would also like to see tests for this. Does this actually... you know... work? 
> 
> So how would you like to continue @jienius? Would you like to get this merged and then focus our attention next to testing, documenting, and refactoring? Or would you rather get that over with right now?
> 
> I feel like we're dragging on our "proof-of-concept" period, and I'd like to see more "production-ready" code. This means tests, documentation, and better factoring!

I will try to address your comments and add more tests. I don't expect this PR to be merged any time soon tbh, there are just so many... uncertainty.

Reply:
I forgot it does fallthrough...